src/browser/suggestion-box.js

```diff
@@ -190,18 +193,94 @@ define((require, exports, module) => {
   // IO.
 
   let xhrSearch;
-  let xhrHistory;
 
 
   Suggestions.reset = suggestions => {
     if (xhrSearch) xhrSearch.abort();
-    if (xhrHistory) xhrHistory.abort();
     return suggestions.clear();
   }
 
+  const dbHistory = new History().stores.pages
+
+  // Calculates the score for use in suggestions from
+  // a result array `match` of `RegExp#exec`.
+  const score = (pattern, input='', base=0.3, length=0.25) => {
+      const index = 1 - base - length
+      const text = String(input);
+      const count = text.length;
+      const match = pattern.exec(text);
+
+      return !match ? -1 :
+              base +
+              length * Math.sqrt(match[0].length / count) +
+              index * (1 - match.index / count);
+  }
+
+  const Pattern = (input, flags="i") => {
```

Comment:
> Maybe pattern should be part of lang or common?

It would make sense to add `lang/parse` or somethig but can we leave it here for now, until we need it elsewhere. Just trying to avoid explosion of modules that we need to fetch before start an app.